apps/opik-backend/src/test/java/com/comet/opik/podam/manufacturer/ProviderApiKeyManufacturer.java

```diff
@@ -0,0 +1,37 @@
+package com.comet.opik.podam.manufacturer;
+
+import com.comet.opik.api.LlmProvider;
+import com.comet.opik.api.ProviderApiKey;
+import org.apache.commons.lang3.RandomStringUtils;
+import uk.co.jemos.podam.api.AttributeMetadata;
+import uk.co.jemos.podam.api.DataProviderStrategy;
+import uk.co.jemos.podam.api.PodamUtils;
+import uk.co.jemos.podam.common.ManufacturingContext;
+import uk.co.jemos.podam.typeManufacturers.AbstractTypeManufacturer;
+
+import java.time.Instant;
+import java.util.UUID;
+
+public class ProviderApiKeyManufacturer extends AbstractTypeManufacturer<ProviderApiKey> {
+    public static final ProviderApiKeyManufacturer INSTANCE = new ProviderApiKeyManufacturer();
+
+    @Override
+    public ProviderApiKey getType(DataProviderStrategy strategy, AttributeMetadata metadata,
+                                 ManufacturingContext context) {
+
+        UUID id = strategy.getTypeValue(metadata, context, UUID.class);
+
+        return ProviderApiKey.builder()
+                .id(id)
+                .provider(randomLlmProvider())
+                .apiKey(RandomStringUtils.randomAlphanumeric(20))
+                .name(strategy.getTypeValue(metadata, context, String.class))
+                .createdBy(strategy.getTypeValue(metadata, context, String.class))
+                .createdAt(Instant.now())
+                .build();
```

Comment:
I don' think you need this manufacturer. If you look at all these fields, this is standard for Podam to generate them randomly.

I believe the issue is that adding the `@Size` annotation broke the Podam generation. That's been the case for other annotations. You would just need to create a `AttributeStrategy` for `@Size.`

Anyway, not a blocker.

Reply:
Actually for `name` (with `@Size` annotation) it works fine. I added this custom manufacturer in order to increase `apiKey` length, default is 10 chars which is not enough. We mask everything less than 12 chars.